mbedtls/build.rs

```diff
@@ -6,32 +6,49 @@
  * option. This file may not be copied, modified, or distributed except
  * according to those terms. */
 
+use std::collections::hash_map::DefaultHasher;
 use std::collections::{HashMap, HashSet};
-use std::env;
+use std::hash::Hasher;
 
 use rustc_version::Channel;
+use std::env;
 
-/// Return the crate hash that Cargo will be passing to `rustc -C metadata=`.
-// If there's a panic in this code block, that means Cargo's way of running the
-// build script has changed, and this code should be updated to handle the new
-// case.
-fn get_compilation_metadata_hash() -> String {
+/// Retrieves or generates a metadata value used for symbol name mangling to ensure unique C symbols.
+/// When building with Cargo, the metadata value is extracted from the OUT_DIR environment variable.
+/// For Bazel builds, this method approximates Cargo's -Cmetadata by hashing the crate name and version,
+/// which are sufficient for ensuring symbol uniqueness.
```

Comment:
Is this correct? What if I have two packages with the same name and version that I both include in my build using package aliasing?

Reply:
> What if I have two packages with the same name and version that I both include in my build using package aliasing?

I don't think that's possible with Bazel. See the paragraph about rules_rust's dependency resolution algorithm in https://github.com/fortanix/rust-mbedtls/pull/365#issuecomment-2293470200.

Reply:
> What about just doing hash($OUT_DIR)?

I suppose that would work because the OUT_DIR contains the package name and version, but it would result in less reproducible/hermetic builds because the hash then contains system-dependent values.

Reply:
@irajtaghlidi can you update this code comment to match the latest approach of the PR?